drivers/pds/lib/resiliency_utils.go

```diff
@@ -0,0 +1,165 @@
+package lib
+
+import (
+	"errors"
+	"sync"
+
+	pds "github.com/portworx/pds-api-go-client/pds/v1alpha1"
+	"github.com/portworx/torpedo/drivers/node"
+
+	_ "github.com/portworx/torpedo/drivers/scheduler/dcos"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/portworx/torpedo/pkg/log"
+	"github.com/portworx/torpedo/tests"
+)
+
+const (
+	active_node_reboot_during_deployment = "active-node-reboot-during-deployment"
+)
+
+// PDS vars
+var (
+	wg                        sync.WaitGroup
+	ResiliencyFlag            = false
+	hasResiliencyConditionMet = false
+	FailureType               TypeOfFailure
+	testError                 error
+	check_till_replica        int32
+	ResiliencyCondition       = make(chan bool)
+)
+
+// Struct Definition for kind of Failure the framework needs to trigger
+type TypeOfFailure struct {
+	Type   string
+	Method func() error
+}
+
+// Wrapper to Define failure type from Test Case
+func DefineFailureType(failuretype TypeOfFailure) {
+	FailureType = failuretype
+}
+
+// Executes all methods in parallel
+func ExecuteInParallel(functions ...func()) {
+	wg.Add(len(functions))
+	defer wg.Wait()
+	for _, fn := range functions {
+		go func(FuncToRun func()) {
+			defer wg.Done()
+			FuncToRun()
+		}(fn)
+	}
+}
+
+// Function to enable Resiliency Test
+func MarkResiliencyTC(resiliency bool) {
+	ResiliencyFlag = resiliency
+	if resiliency {
+		tests.InitInstance()
+	}
+}
+
+// Function to wait for event to induce failure
+func InduceFailure(failure string, ns string) {
+	isResiliencyConditionset := <-ResiliencyCondition
+	if isResiliencyConditionset {
+		FailureType.Method()
+	} else {
+		testError = errors.New("Resiliency Condition did not meet. Failing this test case.")
+		return
+	}
+	return
+}
+
+// Close all open Resiliency channels here
+func CloseResiliencyChannel() {
+	close(ResiliencyCondition)
+}
+
+//
+func InduceFailureAfterWaitingForCondition(deployment *pds.ModelsDeployment, namespace string, CheckTillReplica int32) error {
+	switch FailureType.Type {
+	// Case when we want to reboot a node onto which a deployment pod is coming up
+	case active_node_reboot_during_deployment:
+		check_till_replica = CheckTillReplica
+		log.InfoD("Entering to check if Data service has %v active pods. Once it does, we will reboot the node it is hosted upon.", check_till_replica)
+		func1 := func() {
+			GetPdsSs(deployment.GetClusterResourceName(), namespace, check_till_replica)
+		}
+		func2 := func() {
+			InduceFailure(FailureType.Type, namespace)
+		}
+		ExecuteInParallel(func1, func2)
+		if testError != nil {
+			return testError
+		}
+	}
+	err := ValidateDataServiceDeployment(deployment, namespace)
+	return err
+}
+
+// Reboot the Active Node onto which the application pod is coming up
+func RebootActiveNodeDuringDeployment(ns string) error {
```

Comment:
are you validating this error  any where?

Reply:
Yes, this is getting validated when all functions execute in parallel and proper error is being returned to the test case